xbmc/video/VideoDatabase.cpp

```diff
@@ -1415,52 +1421,77 @@ int CVideoDatabase::GetEpisodeId(const std::string& strFilenameAndPath, int idEp
   {
     if (nullptr == m_pDB)
       return -1;
-    if (nullptr == m_pDS)
-      return -1;
 
     // need this due to the nested GetEpisodeInfo query
     std::unique_ptr<Dataset> pDS;
     pDS.reset(m_pDB->CreateDataset());
     if (nullptr == pDS)
       return -1;
 
-    int idFile = GetFileId(strFilenameAndPath);
-    if (idFile < 0)
-      return -1;
-
-    std::string strSQL=PrepareSQL("select idEpisode from episode where idFile=%i", idFile);
-
-    CLog::LogFC(LOGDEBUG, LOGDATABASE, "({}), query = {}", CURL::GetRedacted(strFilenameAndPath),
-                strSQL);
-    pDS->query(strSQL);
-    if (pDS->num_rows() > 0)
+    const int idFile{GetFileId(strFilenameAndPath)};
+    int idReturnEpisode{-1};
+    if (idFile > 0)
     {
-      if (idEpisode == -1)
-        idEpisode = pDS->fv("episode.idEpisode").get_asInt();
-      else // use the hint!
+      const std::string strSQL{PrepareSQL("select idEpisode from episode where idFile=%i", idFile)};
+
+      CLog::LogFC(LOGDEBUG, LOGDATABASE, "({}), query = {}", CURL::GetRedacted(strFilenameAndPath),
+                  strSQL);
+      pDS->query(strSQL);
+      if (pDS->num_rows() > 0)
```

Comment:
pDS remains open after this block. it's not a good idea to rely on dataset self-protection logic to close automatically when it's used again.